### PR TITLE
Bug 1108955 - Tweak multilocale optimize performances. r=stas

### PR DESCRIPTION
**Pull request by ochameau**
https://bugzilla.mozilla.org/show_bug.cgi?id=1108955

### CODE REVIEW

File: build/webapp-optimize.js [hunk 1]
```diff
@@ -139,18 +139,21 @@ HTMLOptimizer.prototype._proceedLocales = function() {
...
       this.webapp.asts[lang] = [];
     }
-    for (let i = 0; i < this.asts[lang].length; i++) {
+    let asts = this.asts[lang];
+    let webappAsts = this.webapp.asts[lang];
+    for (let i = 0; i < asts.length; i++) {
       let index = -1;
-      for (let j = 0; j < this.webapp.asts[lang].length; j++) {
-        if (this.webapp.asts[lang][j].$i === this.asts[lang][i].$i) {
+      let current = asts[i];
```
Comment (stasm) [inline build/webapp-optimize.js]: How about:

    let identifierToFind = ast[i].$i;

**Comment by try-server-hook** [comment]
[ochameau null (ochameau)](https://github.com/ochameau) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=f6927e9bdd71)

File: build/webapp-optimize.js [hunk 2]
```diff
@@ -139,18 +139,21 @@ HTMLOptimizer.prototype._proceedLocales = function() {
...
-    for (let i = 0; i < this.asts[lang].length; i++) {
+    let asts = this.asts[lang];
+    let webappAsts = this.webapp.asts[lang];
+    for (let i = 0; i < asts.length; i++) {
       let index = -1;
-      for (let j = 0; j < this.webapp.asts[lang].length; j++) {
-        if (this.webapp.asts[lang][j].$i === this.asts[lang][i].$i) {
+      let identifierToFind = ast[i].$i;
+      for (let j = 0; j < webappAsts.length; j++) {
+        if (webappAsts[j].$i === identifierToFind.$i) {
```
Comment (stasm) [inline build/webapp-optimize.js]: You don't need `.$i` here anymore, I think.

Comment (stasm) [inline build/webapp-optimize.js]: I mean in `identiferToFind`.  You still need it on the right hand side.

Comment (ochameau) [inline build/webapp-optimize.js]: Good catch thanks!

**Comment by try-server-hook** [comment]
[ochameau null (ochameau)](https://github.com/ochameau) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=efd5ddbf57fd)

**Comment by try-server-hook** [comment]
[ochameau null (ochameau)](https://github.com/ochameau) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=3b8581d3b444)

**Pull request merged**